backend/routes/stocks.js

```diff
@@ -67,24 +66,27 @@ router.get("/stocks/:ticker", async (req, res) => {
                 where: { UserId, StockId }
             });
 
-            stock.dataValues.following = following != null;
+            const currentStock = { ...stock.dataValues };
+
+            currentStock.following = following != null;
```

Comment:
Well done. You have avoid mutating the stock object. 

Nit: You can even avoid mutations to the new object with:

```
const currentStock = { 
  ...stock.dataValues,
  following: following != null,
  liking: liking != null,
  disliking: disliking != null
};
```